### PR TITLE
Add the PhoneNumberParseService

### PR DESCRIPTION
**Pull request by kbosma123**

### CODE REVIEW

File: src/PhoneNumberParseService.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+use DR\Internationalization\PhoneNumber\PhoneNumberTypeEnum;
+use InvalidArgumentException;
+use libphonenumber\NumberParseException;
+use libphonenumber\PhoneNumberType;
+use libphonenumber\PhoneNumberUtil;
+
+class PhoneNumberParseService
+{
+    private ?PhoneNumberUtil $phoneNumberUtil = null;
+    public function __construct(private string $countryCode)
```
Comment (frankdekker) [inline src/PhoneNumberParseService.php:16]: add readonly property

File: src/PhoneNumber/PhoneNumber.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+class PhoneNumber
+{
+    /**
+     * @param string $countryCode 2 Character ISO country code (NL, BE, FR, ES)
+     *
+     * @internal Only receive this object from the PhoneNumberParseService::parse function.
+     */
+    public function __construct(
+        private string $internationalDialCode,
```
Comment (frankdekker) [inline src/PhoneNumber/PhoneNumber.php:16]: add readonly property

File: src/PhoneNumber/PhoneNumber.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        private string $nationalNumber,
+        private string $rawInput,
+        private string $countryCode,
+        private PhoneNumberTypeEnum $numberType,
+        private LibPhoneNumber $phoneNumber,
+        private ?string $extension = null
+    ) {
+    }
+
+    public function getInternationalDialCode(): string
```
Comment (frankdekker) [inline src/PhoneNumber/PhoneNumber.php:27]: Could you add some docblocks with a description to these methods. What do these values mean? Maybe with an example?

File: src/PhoneNumber/PhoneNumberTypeEnum.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<?php
+declare(strict_types=1);
+
+namespace DR\Internationalization\PhoneNumber;
+
+enum PhoneNumberTypeEnum: string
+{
+    case FIXED_LINE = 'FIXED_LINE';
```
Comment (frankdekker) [inline src/PhoneNumber/PhoneNumberTypeEnum.php:8]: seeing you need to convert twice to go from `numberType` to enum. change this enum to:

`case FIXED_LINE = PhoneNumberType::FIXED_LINE`

and change enum to `PhoneNumberTypeEnum: int`

File: src/PhoneNumberParseService.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            (string)$this->phoneNumberUtil->getRegionCodeForNumber($parsedNumber),
+            $this->getNumberType($this->phoneNumberUtil->getNumberType($parsedNumber)),
+            $parsedNumber,
+            $parsedNumber->getExtension()
+        );
+    }
+
+    private function getNumberType(int $numberType): PhoneNumberTypeEnum
+    {
+        $numberType = PhoneNumberType::values()[$numberType];
```
Comment (frankdekker) [inline src/PhoneNumberParseService.php:56]: With the enum change, this conversion should not be necessary anymore.

File: src/PhoneNumberParseService.php [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+use DR\Internationalization\PhoneNumber\PhoneNumber;
+use DR\Internationalization\PhoneNumber\PhoneNumberTypeEnum;
+use InvalidArgumentException;
+use libphonenumber\NumberParseException;
+use libphonenumber\PhoneNumberType;
+use libphonenumber\PhoneNumberUtil;
+
+class PhoneNumberParseService
+{
+    private ?PhoneNumberUtil $phoneNumberUtil = null;
```
Comment (frankdekker) [inline src/PhoneNumberParseService.php:15]: Add newline after this one.

**Review by frankdekker — CHANGES_REQUESTED** [review]

File: src/PhoneNumber/PhoneNumberTypeEnum.php (hunk 4 above)
Comment (kbosma123) [reply src/PhoneNumber/PhoneNumberTypeEnum.php:8]: As discussed, not possible because some php limits.

**Review by frankdekker — APPROVED** [review]

**Pull request merged**